question-service/app/crud/questions.py

```diff
@@ -39,7 +39,11 @@ async def delete_question(question_id: str):
 async def update_question_by_id(question_id: str, question_data: UpdateQuestionModel):
     existing_question = await question_collection.find_one({"_id": ObjectId(question_id)})
     if existing_question is None:
-        return None
+        return {"error": "does not exist"}
+    
+    duplicate_question = await question_collection.find_one({"title": question_data.title})
+    if duplicate_question:
```

Comment:
This makes it such that if you don't change the title, you cannot update.

Probably will have to check against the id, if same id, then allow the update.